dask_planner/src/sql.rs

```diff
@@ -148,20 +148,20 @@ impl ContextProvider for DaskSQLContext {
 
         match name {
             "year" => {
-                let sig = Signature::variadic(vec![DataType::Int64], Volatility::Immutable);
+                let sig = Signature::exact(vec![DataType::Int64], Volatility::Immutable);
                 let rtf: ReturnTypeFunction = Arc::new(|_| Ok(Arc::new(DataType::Int64)));
                 return Some(Arc::new(ScalarUDF::new(name, &sig, &rtf, &fun)));
             }
             "atan2" | "mod" => {
-                let sig = Signature::variadic(
+                let sig = Signature::exact(
                     vec![DataType::Float64, DataType::Float64],
```

Comment:
Maybe we should allow `Int64` as input here? If we don't, we end up forcing an unnecessary cast to floats:

```python
In [10]: c.explain("select mod(10, 30)")
WARNING:dask_sql.context:No tables are registered.
Skipping optimizer rule simplify_expressions due to unexpected error: This feature is not implemented: 
Skipping optimizer rule simplify_expressions due to unexpected error: This feature is not implemented: 
Skipping optimizer rule simplify_expressions due to unexpected error: This feature is not implemented: 
Out[10]: 'Projection: mod(CAST(Int64(10) AS Float64), CAST(Int64(30) AS Float64))\n  EmptyRelation'
```

This illustrates some of the difficulty I had when trying to switch away from `variadic`, as essentially, we would want to allow any length-2 combination of floats and ints, which can get cumbersome to write out - some workarounds could be making a utility function that can generate a `TypeSignature::OneOf` given some parameters, or maybe even the introduction of a new `TypeSignature` variant similar to `Variadic`, but with a fixed length.

Also, trying out `atan2` I notice that it is seemingly being handled implicitly on the DataFusion end, which seems to imply that we no longer need to define it here:

```python
In [11]: c.explain("select atan2(10, 30)")
Out[11]: 'Projection: Float32(0.32175055) AS atan2(Int64(10),Int64(30))\n  EmptyRelation'
```

Doesn't need to be done in this PR, but might be worthwhile going through the custom UDF/UDAFs to see if any have been implemented upstream and can be removed.

Reply:
I like the idea of making a utility function that generates all combinations of type inputs! I'll try and get it working.

I can also check out which custom functions are no longer needed.